src/provider/CartProvider.tsx

```diff
@@ -24,117 +29,73 @@ type MyContextProviderProps = {
 };
 
 const CartProvider: React.FC<MyContextProviderProps> = ({ children }) => {
-  const [addedIDS, setAddedIDS] = useState<Record<number, number>>({});
-  const [cartValue, setCartValue] = useState<number>(0);
   const [cartItems, setCartItems] = useState<Product[]>([]);
 
+  const cartValue = useMemo(
+    () => cartItems.reduce((prev, curr) => prev + curr.price * curr.price, 0),
+    [cartItems],
+  );
```

Comment:
It displays squared value of price, instead calculate based on price multiplied by quantity